R/data.R

```diff
@@ -108,3 +108,35 @@ use_data_raw <- function(base_path = ".") {
   message("* Add data creation scripts in 'data-raw'")
   message("* Use devtools::use_data() to add data to package")
 }
+
+uses_data <- function(path) {
+  dir.exists(file.path(path, "data"))
+}
+
+#' List the datasets in a package
+#'
+#' A convenience wrapper for \code{\link[utils]{data}} to list the name and
+#' title of all the datasets in an installed package.
+#'
+#' @param pkg A string naming a package.
+#' @return A data frame with two column, with one row per dataset.
+#' \describe{
+#' \item{Item}{Character. The name of the dataset.}
+#' \item{Title}{Character. A sentence describing the dataset.}
+#' }
+#' @seealso \code{\link[utils]{data}}, \code{\link{use_readme_rmd}}
+#' @export
+list_datasets <- function(pkg) {
+  stopifnot(is.character(pkg), length(pkg) == 1)
+  if(!is_installed(pkg)) {
+    no_datasets <- data.frame(
+      Item = character(),
+      Title = character(),
+      stringsAsFactors = FALSE
+    )
+    return(no_datasets)
+  }
+  as.data.frame(
+    utils::data(package = pkg)$results[, c("Item", "Title"), drop = FALSE],
+    stringsAsFactors = FALSE)
```

Comment:
Fixed via [this commit](https://github.com/richierocks/usethis/commit/89da4538b1c6d5b1e5f17b229a8e0688db53d210), but note that it adds `tibble` and `magrittr` dependencies.